docs/en/reference/inheritance-mapping.rst

```diff
@@ -38,38 +38,36 @@ Example:
     use Doctrine\ORM\Mapping\MappedSuperclass;
     use Doctrine\ORM\Mapping\Entity;
 
-    /** @MappedSuperclass */
+    #[MappedSuperclass]
     class Person
     {
-        /** @Column(type="integer") */
-        protected $mapped1;
-        /** @Column(type="string") */
-        protected $mapped2;
-        /**
-         * @OneToOne(targetEntity="Toothbrush")
-         * @JoinColumn(name="toothbrush_id", referencedColumnName="id")
-         */
-        protected $toothbrush;
+        #[Column(type: 'integer')]
+        protected int $mapped1;
+        #[Column(type: 'string')]
+        protected string $mapped2;
+        #[OneToOne(targetEntity: Toothbrush::class)]
+        #[JoinColumn(name: 'toothbrush_id', referencedColumnName: 'id')]
+        protected Toothbrush|null $toothbrush = null;
 
         // ... more fields and methods
     }
-    
-    /** @Entity */
+
+    #[Entity]
     class Employee extends Person
     {
-        /** @Id @Column(type="integer") */
-        private $id;
-        /** @Column(type="string") */
-        private $name;
-    
+        #[Id, Column(type: 'integer')]
+        private int|null $id = null;
+        #[Column(type: 'string')]
+        private string $name;
```

Comment:
Probably should provide a default value, otherwise: creating new entity and calling getName will return fatal - if getter is strict too.

Reply:
On the other hand, you might have a constructor preventing this entirely.